model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -81,6 +81,13 @@ service BeamFnControl {
       returns (
           // A stream of instructions requested of the SDK to be performed.
           stream InstructionRequest) {}
+
+  rpc GetProcessBundleDescriptor(GetProcessBundleDescriptorRequest) returns (
```

Comment:
I would have expected this to be a separate stream containing SDK requests similar to how Control works or made Control bidirectional for requests and responses by making a singular `Instruction` field that is streamed both ways that contains both Request and Response fields.

The advantage of using a stream is that it may be important to provide ordering over SDK requests/responses. Putting it into the Control stream allows for a total ordering between all Runner and SDK requests/responses (to solve the instruction id collision issue, we can assign servers the "s" prefix and clients the "c" prefix).

Reply:
The current Control must be a stream because (1) it is server-initiated requests on a client-initiated connection and (2) in streaming pipelines, it we want it to be high-volume, low-latency, possibly pipelined. However, using streams in this way vs. "standard" RPC methods comes at a cost (in both code and conceptual complexity). 

As far as ordering, it is preferable to have a protocol that is not sensitive to this kind of thing. (In fact, this eliminates the one need for ordering in our Control stream, where bundles needed to be registered before they were used (and this didn't fit well with pattern of dispatching instructions to threadpool-backed handlers in the SDKs which required extra locking/waiting anyway). Progress and splitting requests were already asynchronous.)

Reply:
sg